src/components/Camera.vue

```diff
@@ -76,6 +77,16 @@ export default defineComponent({
             const devices = await navigator.mediaDevices.enumerateDevices();
             return devices.filter((device) => kinds.includes(device.kind));
         };
+        
+        const currentDevice = (stream: MediaStream | undefined) => {
+            if (!stream) return;
+            const used_devices = stream
+                .getVideoTracks()
+                .map((track) => track.getSettings().deviceId);
+            if (used_devices.length === 1) {
+                emit("used-video-device", used_devices[0]);
```

Comment:
Doesn't the change camera event not already tell which new device id is active? I would make this function return the used device id and remove the event. Then you could retrieve the value with:

```vue
const camera = ref<InstanceType<typeof Camera>>();

const currentDeviceID = camera.value.currentDevice();
```

Also, is the first entry in the `used_devices` array always from the active device?